clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -706,8 +705,18 @@ public CompletedFetch(TopicPartition partition,
         private final FetchManagerMetrics sensors;
         private final Set<TopicPartition> unrecordedPartitions;
 
-        private int totalBytes;
-        private int totalRecords;
+        private static class FetchMetrics {
```

Comment:
nitpick: could we move this below the record() function? It would be nice to keep the class fields together.